### PR TITLE
use flyway

### PR DESCRIPTION
**Pull request by leevccc**
## about change

1. change all table charset to utf8mb4
2. table `savedlocations` add type `JAIL` (JAIL command have some problem about warp function, 'JAIL' is for that, and mayby i will fix it)
3. switch to flyway manage sql version
4. add a config about flyway, default false, it will not to init database, but still update database in the future

new folder

1. flyway/initial

Every file in this folder is based on the database file and is used for initializing database tables. Please do not make any changes. If you have already initialized the database, simply set INIT_DATABASE_AT_FIRST_START in the config.yaml file to false, and everything will be okay.

2. flyway/community

If you wish to modify the database, please place your SQL scripts in this folder.

3. flyway/customize

If you want to modify the database and keep the changes private, please store your SQL scripts in this folder.

## why use flyway

In short, it can help us better maintain the database. When we start the service, the database will be updated. We no longer have to check what updates are in the sql file, and then write corresponding statements to execute in the database.

more about you can see /flyway/README.md

### CODE REVIEW

**Comment by P0nk** [comment]
Introducing Flyway is good. I'm working on it in https://github.com/P0nk/Cosmic/pull/161 but it's not done yet so this is a welcome PR. I will likely make modifications to it down the road.

However, please refrain from adding more features in the same PR. The "level up protect" and jail warp fix don't belong in the same PR as the Flyway stuff.

**Comment by leevccc** [comment]
> Introducing Flyway is good. I'm working on it in #161 but it's not done yet so this is a welcome PR. I will likely make modifications to it down the road.
> 
> However, please refrain from adding more features in the same PR. The "level up protect" and jail warp fix don't belong in the same PR as the Flyway stuff.

Sorry, I'm learning to use PR, I've reverted those two commits on my branch
